src/jit/codegenarm64.cpp

```diff
@@ -2664,6 +2664,11 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* cpBlkNode)
 
     emitter* emit = getEmitter();
 
+    if (cpBlkNode->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
```

Comment:
A comment stating that we place a full memory barrier before and after the unrolled operation, should be added here